### ISSUE TITLE
Drop dependency to gtkdoc-mktmpl

### ISSUE THREAD

**Issue by kenhys** [clear-code/cutter#33]
## Problem

gtkdoc-mktmpl is removed since gtk-doc-tools 1.26.
Cutter rely on it to generate documents.

## Expected

`make -C doc` generates documents correctly without gtkdoc-mktmpl.

## Note

`make -C doc` still works on Debian stretch.

**Comment by kenhys** [comment]
See https://developer.gnome.org/gtk-doc-manual/stable/modernizing.html.en
for migration.

**Comment by kenhys** [comment]
Here is the check list about affected or not.

debian jessie: gtk-doc-tools 1.21
debian stretch: gtk-doc-tools 1.25
debian unstable: gtk-doc-tools 1.26 (gtkdoc-mktmpl removed)

ubuntu trusty(14.04): gtk-doc-tools 1.20
ubuntu zenial(16.04): gtk-doc-tools 1.25
ubuntu zesty(17.04): gtk-doc-tools 1.25
ubuntu artful(17.10): gtk-doc-tools 1.25

**Comment by kenhys** [comment]
It is good enough to support GTK-Doc 1.9?

https://developer.gnome.org/gtk-doc-manual/stable/modernizing-gtk-doc-1-9.html.en

**Comment by kenhys** [comment]
It seems that goocanvas-2.0 already fixed similar issue.

https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=876597

**Comment by kenhys** [comment]
In goocanvas 2.0 case, it raised from 1.8 to 1.16 at once.

**Comment by kenhys** [comment]
https://developer.gnome.org/gtk-doc-manual/stable/settingup_autoconf.html.en

It is seems that additional configuration is needed for cutter.

```
diff --git a/configure.ac b/configure.ac
index 181b169b..6fe0f3d5 100644
--- a/configure.ac
+++ b/configure.ac
@@ -555,7 +555,7 @@ dnl Checks for gtk-doc
 dnl **************************************************************
 
 m4_ifdef([GTK_DOC_CHECK],
-         [GTK_DOC_CHECK([1.8])],
+         [GTK_DOC_CHECK([1.8],[--flavour no-tmpl])],
          [enable_gtk_doc=no
           AM_CONDITIONAL([GTK_DOC_USE_LIBTOOL], [false])
           AM_CONDITIONAL([ENABLE_GTK_DOC], [false])])
```

Only `--flavour no-tmpl` configuration is not enough.

**Comment by kenhys** [comment]
It has nothing to do with `configure.ac` configuration. It is affected by `gtk-doc.make`.

**Comment by kenhys** [comment]
`/usr/share/gtk-doc/data/gtk-doc.make` is the latest version. but there is no support to i18n.

**Comment by kenhys** [comment]
current gtk-doc.make use `gtkdoc-mktmpl` and it is included from `doc/reference/Makefile.am`.

**Comment by kenhys** [comment]
Just updating `gtk-doc.make` breaks because `$lang` is not considered.

**Comment by kenhys** [comment]
Merged https://github.com/clear-code/cutter/pull/34 before updating `gtk-doc.make`. some legacy parts are removed.

**Comment by kenhys** [comment]
Instead of updating gtk-doc.make manually, it seems that `gtkdocize --copy` & overriding rules by Makefile.am is better solution.

**Comment by kenhys** [comment]
Note that `gtkdocize` doesn't care the case that `GTK_DOC_CHECK` is used with `m4_ifdef`.

ref. https://github.com/GNOME/gtk-doc/blob/master/gtkdocize.in#L60

 so as a workaround, indent of `GTK_DOC_CHECK` must be modified a bit.
There is a example about `configure.ac` of milter-manager project. 

ref. https://github.com/milter-manager/milter-manager/blob/master/configure.ac#L680

**Comment by kenhys** [comment]
> Note that gtkdocize doesn't care the case that GTK_DOC_CHECK is used with m4_ifdef.

Merged commit which supports `gtkdocize`.
https://github.com/clear-code/cutter/pull/36

**Comment by kenhys** [comment]
fixed with #37
